<a href="https://colab.research.google.com/github/JoaoSchneiderr/Contagem-para-metas/blob/main/notebooks/Getting_started_with_google_colab_ai.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importação de informações

In [2]:
!pip install gdown
!gdown --id 1GlMLuPx9CLWjJo9iOG0JHvz7ch_dTGee -O Dados.csv
import pandas as pd

/usr/local/lib/python3.12/dist-packages/gdown/__main__.py:139: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(
Downloading...
From: https://drive.google.com/uc?id=1GlMLuPx9CLWjJo9iOG0JHvz7ch_dTGee
To: /content/Dados.csv
100% 607k/607k [00:00<00:00, 34.4MB/s]


In [13]:
nova_iguacu_df = pd.read_csv('Dados.csv', encoding='latin1')
display(nova_iguacu_df.head())

,id,pesid,data_inversa,dia_semana,horario,uf,br,km,municipio,causa_principal,...,estado_fisico,idade,sexo,ilesos,feridos_leves,feridos_graves,mortos,latitude,longitude,faixa_etaria
0,653257,1469809.0,2025-01-04,sábado,23:00:00,RJ,493.0,66,NOVA IGUACU,Não,...,Ileso,53.0,Masculino,1.0,0.0,0.0,0.0,-22.682450,-43.453331,Adulto
1,653257,1469811.0,2025-01-04,sábado,23:00:00,RJ,493.0,66,NOVA IGUACU,Não,...,Óbito,33.0,Masculino,0.0,0.0,0.0,1.0,-22.682450,-43.453331,Adulto
2,653257,1469809.0,2025-01-04,sábado,23:00:00,RJ,493.0,66,NOVA IGUACU,Sim,...,Ileso,53.0,Masculino,1.0,0.0,0.0,0.0,-22.682450,-43.453331,Adulto
3,653257,1469811.0,2025-01-04,sábado,23:00:00,RJ,493.0,66,NOVA IGUACU,Sim,...,Óbito,33.0,Masculino,0.0,0.0,0.0,1.0,-22.682450,-43.453331,Adulto
4,653414,1470243.0,2025-01-05,domingo,22:50:00,RJ,116.0,181,NOVA IGUACU,Sim,...,Ileso,47.0,Masculino,1.0,0.0,0.0,0.0,-22.761312,-43.423492,Adulto


### 1. Quantidade total de acidentes em Nova Iguaçu / Quantidade de acidentes registrados por mês


In [11]:
acidentes_total = len(nova_iguacu_df)
acidentes_total

1807

In [48]:
acidentes_por_mes = nova_iguacu_df.copy()
acidentes_por_mes['data_inversa'] = pd.to_datetime(acidentes_por_mes['data_inversa'])

meses_pt = {
    1: 'Janeiro', 2: 'Fevereiro', 3: 'Março', 4: 'Abril',
    5: 'Maio', 6: 'Junho', 7: 'Julho', 8: 'Agosto',
    9: 'Setembro', 10: 'Outubro', 11: 'Novembro', 12: 'Dezembro'
}

contagem_mes = acidentes_por_mes['data_inversa'].dt.month.value_counts().sort_index()
contagem_mes.index.name = 'Divisão por mês'
display(contagem_mes.rename(index=meses_pt))

,count
Divisão por mês,
Janeiro,97
Fevereiro,153
Março,179
Abril,204
Maio,115
Junho,192
Julho,107
Agosto,164
Setembro,108


### 2. Gravidade dos acidentes por mês em Nova Iguaçu


In [59]:
print("\n Análise da Gravidade dos Acidentes por Mês em Nova Iguaçu (Média Mensal):")
avg_mortos_mes = gravidade_mensal['mortos'].mean()
avg_feridos_graves_mes = gravidade_mensal['feridos_graves'].mean()
avg_feridos_leves_mes = gravidade_mensal['feridos_leves'].mean()
print(f"Em média, mensalmente, ocorreram {avg_mortos_mes:.1f} mortes, {avg_feridos_graves_mes:.1f} feridos graves e {avg_feridos_leves_mes:.1f} feridos leves.")

total_mortos_geral = gravidade_mensal['mortos'].sum()
total_feridos_graves_geral = gravidade_mensal['feridos_graves'].sum()
total_feridos_leves_geral = gravidade_mensal['feridos_leves'].sum()
total_ilesos_geral = gravidade_mensal['ilesos'].sum()

print("\n Totais Gerais e Proporções (Ano de 2025) ")
total_mortos = total_mortos_geral
total_feridos = total_feridos_graves_geral + total_feridos_leves_geral
print(f"Total Geral de Mortos: {total_mortos:.0f}")
print(f"Total Geral de Feridos (Graves + Leves): {total_feridos:.0f}")


 Análise da Gravidade dos Acidentes por Mês em Nova Iguaçu (Média Mensal):
Em média, mensalmente, ocorreram 3.2 mortes, 7.8 feridos graves e 61.9 feridos leves.

 Totais Gerais e Proporções (Ano de 2025) 
Total Geral de Mortos: 39
Total Geral de Feridos (Graves + Leves): 836


In [60]:
classificacao_acidente_contagem = nova_iguacu_df['classificacao_acidente'].value_counts()

acidentes_com_vitimas_fatal = classificacao_acidente_contagem.get('Com Vítimas Fatais', 0)
acidentes_com_vitimas_feridas = classificacao_acidente_contagem.get('Com Vítimas Feridas', 0)
acidentes_sem_vitimas = classificacao_acidente_contagem.get('Sem Vítimas', 0)

total_registros_acidentes = len(nova_iguacu_df)

prop_com_vitimas = ((acidentes_com_vitimas_fatal + acidentes_com_vitimas_feridas) / total_registros_acidentes) * 100
prop_sem_vitimas = (acidentes_sem_vitimas / total_registros_acidentes) * 100

print(f"Proporção de registros de acidentes com vítimas (fatais ou feridas): {prop_com_vitimas:.2f}%")
print(f"Proporção de registros de acidentes sem vítimas: {prop_sem_vitimas:.2f}%")

Proporção de registros de acidentes com vítimas (fatais ou feridas): 92.03%
Proporção de registros de acidentes sem vítimas: 7.97%


In [61]:
taxa_acidentes_morte = (acidentes_com_vitimas_fatal / total_registros_acidentes) * 100
print(f"Taxa de registros de acidentes com vítimas fatais: {taxa_acidentes_morte:.2f}%")

Taxa de registros de acidentes com vítimas fatais: 6.64%


In [63]:
taxa_acidentes_morte = (acidentes_com_vitimas_fatal / total_registros_acidentes) * 100
print(f"Taxa de registros de acidentes com vítimas fatais: {taxa_acidentes_morte:.2f}%")


total_unique_accidents = nova_iguacu_df['id'].nunique()
total_vitimas = total_mortos_geral + total_feridos_graves_geral + total_feridos_leves_geral
media_vitimas_por_acidente = total_vitimas / total_unique_accidents
print(f"Média de vítimas (mortos + feridos graves + feridos leves) por acidente único: {media_vitimas_por_acidente:.2f}")

Taxa de registros de acidentes com vítimas fatais: 6.64%
Média de vítimas (mortos + feridos graves + feridos leves) por acidente único: 2.00


### 3. Tipos de acidentes


In [77]:
print("\n Tipos de Acidentes Mais Comuns")
tipos_acidentes_comuns = nova_iguacu_df['tipo_acidente'].value_counts().head(10)
display(tipos_acidentes_comuns)


 Tipos de Acidentes Mais Comuns


,count
tipo_acidente,
Colisão traseira,487
Queda de ocupante de veículo,351
Colisão lateral mesmo sentido,270
Tombamento,166
Colisão transversal,113
Saída de leito carroçável,79
Colisão frontal,64
Colisão com objeto,63
Atropelamento de Pedestre,62


In [78]:
print("\n Classificação de Acidentes Mais Comuns")
classificacao_acidente_comum = nova_iguacu_df['classificacao_acidente'].value_counts()
display(classificacao_acidente_comum)


 Classificação de Acidentes Mais Comuns


,count
classificacao_acidente,
Com Vítimas Feridas,1543
Sem Vítimas,144
Com Vítimas Fatais,120


### 4. Veículos envolvidos


In [ ]:
print("\n Top 10 Tipos de Veículos Mais Comuns nos Acidentes em Nova Iguaçu ")
tipos_veiculo_comuns = nova_iguacu_df['tipo_veiculo'].value_counts().head(10)
display(tipos_veiculo_comuns)


--- Top 10 Tipos de Veículos Mais Comuns nos Acidentes em Nova Iguaçu ---


,count
tipo_veiculo,
Motocicleta,703
Automóvel,666
Caminhonete,93
Caminhão,69
Camioneta,60
Caminhão-trator,52
Semireboque,39
Desconhecido,37
Motoneta,23


In [6]:
print("\n CARRO VS MOTO VS CAMINHÃO")
veiculo_especifico = nova_iguacu_df['tipo_veiculo'].value_counts().loc[['Carro', 'Motocicleta', 'Caminhão']]
display(veiculo_especifico)


 CARRO VS MOTO VS CAMINHÃO


NameError: name 'nova_iguacu_df' is not defined

### 5. Perfil das pessoas envolvidas


In [16]:
df_perfil = nova_iguacu_df[['sexo', 'idade', 'estado_fisico', 'tipo_envolvido']].copy()
df_perfil['idade'] = pd.to_numeric(df_perfil['idade'], errors='coerce')
df_perfil = df_perfil.dropna(subset=['idade'])
display(df_perfil.head())

,sexo,idade,estado_fisico,tipo_envolvido
0,Masculino,53.0,Ileso,Condutor
1,Masculino,33.0,Óbito,Pedestre
2,Masculino,53.0,Ileso,Condutor
3,Masculino,33.0,Óbito,Pedestre
4,Masculino,47.0,Ileso,Condutor


In [17]:
bins = [0, 18, 25, 35, 45, 55, 65, 100]
labels = ['0-17', '18-24', '25-34', '35-44', '45-54', '55-64', '65+']

df_perfil['faixa_etaria'] = pd.cut(df_perfil['idade'], bins=bins, labels=labels, right=False)

print("\nDistribuição por Faixa Etária (Indivíduos Envolvidos em Acidentes):")
display(df_perfil['faixa_etaria'].value_counts().sort_index())


Distribuição por Faixa Etária (Indivíduos Envolvidos em Acidentes):


,count
faixa_etaria,
0-17,25
18-24,162
25-34,472
35-44,420
45-54,293
55-64,148
65+,60


In [18]:
df_vitimas = df_perfil[df_perfil['estado_fisico'].isin(['Ferido Grave', 'Ferido Leve', 'Óbito'])]
idade_media_vitimas = df_vitimas['idade'].mean()

print(f"Idade média das vítimas (Ferido Grave, Ferido Leve ou Morto): {idade_media_vitimas:.2f} anos")

Idade média das vítimas (Ferido Grave, Ferido Leve ou Morto): 28.92 anos


In [19]:
print("\nDistribuição por Sexo (Indivíduos Envolvidos em Acidentes):")
contagem_sexo = df_perfil['sexo'].value_counts()
display(contagem_sexo)

porcentagem_sexo = df_perfil['sexo'].value_counts(normalize=True) * 100
print("\nPorcentagem por Sexo (Indivíduos Envolvidos em Acidentes):")
display(porcentagem_sexo)


Distribuição por Sexo (Indivíduos Envolvidos em Acidentes):


,count
sexo,
Masculino,1309
Feminino,312
Não Informado,179
Ignorado,7



Porcentagem por Sexo (Indivíduos Envolvidos em Acidentes):


,proportion
sexo,
Masculino,72.440509
Feminino,17.266187
Não Informado,9.905921
Ignorado,0.387382


In [20]:
print("\nContagem de Condutores vs. Passageiros:\n")
contagem_condutor_passageiro = df_perfil['tipo_envolvido'].value_counts().loc[['Condutor', 'Passageiro']]
display(contagem_condutor_passageiro)


Contagem de Condutores vs. Passageiros:



,count
tipo_envolvido,
Condutor,1361
Passageiro,363


### 6. Distribuição temporal


In [21]:
print("\n Acidentes por Horário em Nova Iguaçu")
acidentes_por_horario = nova_iguacu_df['horario'].value_counts().sort_values(ascending=False)
display(acidentes_por_horario)



 Acidentes por Horário em Nova Iguaçu


,count
horario,
17:00:00,53
02:50:00,51
08:00:00,41
11:45:00,37
20:30:00,36
...,...
22:05:00,1
22:32:00,1
15:50:00,1


In [22]:
print("\n--- Acidentes por Dia da Semana em Nova Iguaçu ---")
acidentes_por_dia_semana = nova_iguacu_df['dia_semana'].value_counts().sort_values(ascending=False)
display(acidentes_por_dia_semana)


--- Acidentes por Dia da Semana em Nova Iguaçu ---


,count
dia_semana,
domingo,387
segunda-feira,320
sexta-feira,272
terça-feira,245
sábado,232
quarta-feira,222
quinta-feira,129


In [23]:
print("\n Acidentes por Condição Meteorológica em Nova Iguaçu")
acidentes_por_condicao_metereologica = nova_iguacu_df['condicao_metereologica'].value_counts().sort_values(ascending=False)
display(acidentes_por_condicao_metereologica)


 Acidentes por Condição Meteorológica em Nova Iguaçu


,count
condicao_metereologica,
Céu Claro,1233
Nublado,233
Chuva,132
Sol,112
Garoa/Chuvisco,75
Ignorado,22


In [24]:
print("\n Acidentes por Fase do Dia em Nova Iguaçu ")
acidentes_por_fase_dia = nova_iguacu_df['fase_dia'].value_counts()
display(acidentes_por_fase_dia)


 Acidentes por Fase do Dia em Nova Iguaçu 


,count
fase_dia,
Pleno dia,1111
Plena Noite,556
Amanhecer,87
Anoitecer,53


In [25]:
print("\n Gravidade dos Acidentes por Fase do Dia em Nova Iguaçu ")
gravidade_por_fase_dia = nova_iguacu_df.groupby('fase_dia')[['mortos', 'feridos_graves', 'feridos_leves']].sum()
display(gravidade_por_fase_dia)



 Gravidade dos Acidentes por Fase do Dia em Nova Iguaçu 


,mortos,feridos_graves,feridos_leves
fase_dia,,,
Amanhecer,5.0,4.0,31.0
Anoitecer,0.0,2.0,25.0
Plena Noite,25.0,32.0,224.0
Pleno dia,9.0,55.0,463.0


### 7. Condições do ambiente



In [26]:
print("\n Acidentes por Tipo de Pista em Nova Iguaçu ")
acidentes_por_tipo_pista = nova_iguacu_df['tipo_pista'].value_counts()
display(acidentes_por_tipo_pista)


 Acidentes por Tipo de Pista em Nova Iguaçu 


,count
tipo_pista,
Dupla,1119
Simples,461
Múltipla,227


In [28]:
print("\n Acidentes por Uso do Solo em Nova Iguaçu")
acidentes_por_uso_solo = nova_iguacu_df['uso_solo'].value_counts()
display(acidentes_por_uso_solo)


 Acidentes por Uso do Solo em Nova Iguaçu


,count
uso_solo,
Sim,1208
Não,599


In [29]:
print("\n Acidentes por Condição Meteorológica em Nova Iguaçu")
acidentes_por_condicao_metereologica = nova_iguacu_df['condicao_metereologica'].value_counts()
display(acidentes_por_condicao_metereologica)


 Acidentes por Condição Meteorológica em Nova Iguaçu


,count
condicao_metereologica,
Céu Claro,1233
Nublado,233
Chuva,132
Sol,112
Garoa/Chuvisco,75
Ignorado,22


In [34]:
print("\n Acidentes por Traçado da Via em Nova Iguaçu ")
acidentes_por_tracado_via = nova_iguacu_df['tracado_via'].value_counts()
display(acidentes_por_tracado_via)


 Acidentes por Traçado da Via em Nova Iguaçu 


,count
tracado_via,
Reta,1626
Reta;Em Obras,72
Curva,63
Interseção de Vias;Curva,12
Em Obras;Reta,10
Reta;Interseção de Vias,6
Aclive;Reta,4
Interseção de Vias,3
Reta;Aclive,3


### 8. Localização dos acidentes

In [31]:
print("\n Acidentes por Rodovia (BR) em Nova Iguaçu ")
acidentes_por_br = nova_iguacu_df['br'].value_counts()
display(acidentes_por_br)


 Acidentes por Rodovia (BR) em Nova Iguaçu 


,count
br,
116.0,1159
465.0,430
493.0,217
40.0,1


In [32]:
print("\n Top 10 Trechos de Rodovia (BR - KM) com mais Acidentes ")
nova_iguacu_df['br_km'] = nova_iguacu_df['br'].astype(str) + '-' + nova_iguacu_df['km'].astype(str)
acidentes_por_br_km = nova_iguacu_df['br_km'].value_counts().head(10)
display(acidentes_por_br_km)


--- Top 10 Trechos de Rodovia (BR - KM) com mais Acidentes ---


,count
br_km,
116.0-186,172
465.0-20,143
116.0-184,137
116.0-185,122
465.0-21,120
116.0-183,106
116.0-182,90
116.0-181,86
116.0-187,75


In [33]:
print("\n Análise de Pontos Críticos por Latitude e Longitude (Top 10 mais frequentes)")
nova_iguacu_df['lat_lon'] = nova_iguacu_df['latitude'].astype(str) + ',' + nova_iguacu_df['longitude'].astype(str)
acidentes_por_lat_lon = nova_iguacu_df['lat_lon'].value_counts().head(10)
display(acidentes_por_lat_lon)


--- Análise de Pontos Críticos por Latitude e Longitude (Top 10 mais frequentes) ---


,count
lat_lon,
"-22.7371434,-43.4578243",45
"-22.846828,-43.605339",38
"-22.7482422,-43.4443263",36
"-22.84934285,-43.603184",34
"-22.73962339,-43.46276495",30
"-22.74151051,-43.45351823",29
"-22.82057068,-43.62160206",28
"-22.84750572,-43.6046505",27
"-22.68140623,-43.53034051",24


### 9. Análise do Acidente Mais Grave

### Conclusão: Um Retrato dos Acidentes em Nova Iguaçu (2025)

A análise aprofundada dos dados de acidentes em Nova Iguaçu no ano de 2025 revela um cenário com **1807 acidentes** registrados, apresentando picos de ocorrências no mês de novembro e, geralmente, uma maior frequência aos domingos e segundas-feiras.

**Gravidade e Vítimas:**
No total, os acidentes resultaram em **39 mortos**, **93 feridos graves** e **743 feridos leves**. A maior parte dos acidentes são classificados como 'Com Vítimas Feridas' (1543), enquanto 'Com Vítimas Fatais' representam uma parcela menor (120). Fevereiro, Junho e Dezembro se destacam com os maiores números de vítimas (mortos, feridos graves e leves).

**Tipos de Acidentes e Veículos Envolvidos:**
Os **tipos de acidentes mais comuns** são **colisão traseira**, **queda de ocupante de veículo** e **colisão lateral no mesmo sentido**. Curiosamente, esses mesmos tipos de acidentes também estão entre os que causam o **maior número total de vítimas**, sugerindo que, além de frequentes, são significativamente impactantes. Os **veículos mais comumente envolvidos** são automóveis e motocicletas, sendo os **carros** os mais recorrentes, seguidos por motocicletas e caminhões. Há uma concentração de acidentes envolvendo motocicletas.

**Perfil dos Envolvidos:**
O perfil dos envolvidos nos acidentes mostra que a **faixa etária de 25 a 34 anos** é a mais atingida, seguida de perto pela faixa de 35 a 44 anos. A **idade média das vítimas** (feridos graves, leves ou mortos) é de aproximadamente **28.92 anos**. Há uma predominância **masculina** nos envolvidos (72.44%), com a maioria sendo **condutores** (1361) em comparação com passageiros (363).

**Distribuição Temporal e Condições Ambientais:**
Os acidentes são mais frequentes em **pleno dia**, mas a **plena noite** registra um número significativamente maior de mortos e feridos. Os horários de pico variam, mas o período noturno e início da manhã apresentam um número relevante de ocorrências. A maioria dos acidentes ocorre sob **céu claro**, indicando que as condições climáticas adversas não são o principal fator desencadeante para a maioria dos acidentes, mas sim outros fatores como o comportamento humano ou a infraestrutura. A maioria das ocorrências se dá em **pista dupla** e em trechos de **reta**.

**Localização dos Acidentes:**
A **BR 116.0** e a **BR 465.0** são as rodovias com o maior número de acidentes. Os trechos críticos incluem **BR 116.0-186** e **BR 465.0-20**, que registram os maiores volumes de ocorrências. Existem também **pontos críticos específicos de latitude e longitude**, como '-22.7371434,-43.4578243', que merecem atenção especial devido à alta frequência de acidentes.

**O Acidente Mais Grave:**
Considerando o maior número de envolvidos/registros no dataset, o acidente mais notável (ID **696680**) ocorreu em **08 de junho de 2025**, na **BR 116, KM 193**, sob **céu claro**, sendo classificado como uma 'Colisão com objeto'. Este acidente registrou 1 morto, 0 feridos graves e 0 feridos leves, mas se destaca por ter o maior número de registros de envolvidos (45). É importante notar que houve múltiplos acidentes com 1 morto e 1 ferido, sendo comum que a gravidade esteja distribuída por vários eventos de menor escala, mas de alto impacto individual.

**Implicações:**
Esses dados sugerem que, embora a maioria dos acidentes ocorra em condições aparentemente normais (céu claro, pista dupla, reta), a concentração em certos horários (noturnos) e trechos de rodovia, somada aos tipos de acidentes e ao perfil dos envolvidos, aponta para a necessidade de campanhas de conscientização sobre segurança no trânsito e possíveis melhorias de infraestrutura em pontos e trechos específicos de Nova Iguaçu, especialmente para condutores jovens e do sexo masculino.